### DISCUSSION TITLE
Want arbitrary SQL as default values: e.g. UTC Unix Timestamps as BigInt Default values  in PostGreSQL

### DISCUSSION THREAD

**Discussion by thirdreplicator** [prisma/prisma#18416] [Q&A] [answered]
## Problem

I'm using PostGreSQL, and I would like default `created_at` BIGINT values to be set to "extract(epoch from now()". I tried using dbgenerated("extract(epoch from now()") but got the  following error.
```
$ npm run migrate:dev

> ft-backend@1.0.0 migrate:dev
> dotenv -e .env.development -- npx prisma migrate dev

Prisma schema loaded from prisma/schema.prisma
Error: Prisma schema validation - (get-config wasm)
Error code: P1012
error: Error validating: This line is not a valid field or attribute definition.
  -->  schema.prisma:43
   | 
42 |   active         Boolean  @default(true)
43 |   created_at    BigInt @default(dbgenerated("extract(epoch from now()")))
44 | 
   | 

Validation Error Count: 1
[Context: getConfig]

Prisma CLI Version : 4.11.0
```
## Suggested solution

I would like the   following syntax to work:
```
created_at    BigInt    @default(dbgenerated("extract(epoch from now()")))
```
In otherwords, it should  generate a column that looks like this in PostGreSQL:
```
created_at bigint DEFAULT EXTRACT(epoch FROM now())
```

**Comment by janpio** [comment]
Your code has 2 opening `(` in front of the first `"`, and then 3 closing ones after the second `"`. (Also inside the `" "` something is off respectively).)

**Reply by nurul3101** [reply]
@thirdreplicator Did you had a chance to look at Jan's response?
Let us know in case you still need assistance. It would be great if you could mark this Discussion as answered to indicate that it has been resolved.

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by thirdreplicator** [reply]
created_at    BigInt    @default(dbgenerated("extract(epoch from now())"))

How about that?

**Reply by nurul3101** [reply, marked as answer]
I tried it and it worked.

This is the model
```prisma
model Account {
  id         String  @id @default(cuid())
  userId     String? @unique
  User       User?
  created_at BigInt  @default(dbgenerated("extract(epoch from now())"))
}
```

And here's the generated record:

<img width="854" alt="Screenshot 2023-04-20 at 1 45 29 AM" src="https://user-images.githubusercontent.com/13049676/233189695-570b247b-a551-4d26-a9a5-f5d16925b762.png">

**Comment by thirdreplicator** [comment]
Oh wow. Nice. I didn't know it could do that. Thank you!

**Reply by nurul3101** [reply]
No worries! :)

**Comment by jason-allen-oneal** [comment]
is there an equivalent to this for MySQL? I'm having major issues trying to get epoch times to insert.

**Reply by luizmb7** [reply]
Hi my friend, I managed to do it in MariaDB 10.3 like this:

`created_at  Int    @default(dbgenerated("unix_timestamp(current_timestamp())"))`

**Reply by ashaffah** [reply]
thank you, it worked

```js
created_at       BigInt            @default(dbgenerated("UNIX_TIMESTAMP(CURRENT_TIMESTAMP())")) @db.BigInt
updated_at       BigInt?           @default(dbgenerated("NULL ON UPDATE UNIX_TIMESTAMP(CURRENT_TIMESTAMP())")) @db.BigInt
```
